### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by richtier**
[this task](https://uktrade.atlassian.net/browse/TT-238)

Early access PR. Do not merge. Finishing off the tests.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+version: 2
+jobs:
+  test:
+    docker:
+      - image: circleci/python:3.5.5
```
Comment (sdonk) [inline .circleci/config.yml]: Probably worth upgrading to Python 3.6

File: directory_forms_api_client/backends.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    @abc.abstractmethod
+    def serialize_data(self, data):
+        return {}
+
+    def save(self, data):
+        serialized_data = self.serialize_data(data)
+        return self.client.submit_generic(serialized_data)
+
+
+class DirectoryFormsBackendEmail(DirectoryFormsBackendBase):
```
Comment (konradko) [inline directory_forms_api_client/backends.py]: `APIFormsEmailBackend` - more common

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    with open('directory_forms_api_client/version.py', 'rb') as src:
+        return str(ast.literal_eval(
+            pattern.search(src.read().decode('utf-8')).group(1)
+        ))
+
+
+def get_requirements():
+    return [str(r.req) for r in list(parse_requirements(
+        'requirements.txt',
+        session=pip.download.PipSession()
```
Comment (konradko) [inline setup.py]: doesn't work with newer versions of pip - see sigauth - we might need to hardcode the reqs or find a new method of loading from the txt

Comment (richtier) [reply setup.py]: this is true for all of our libs. lets do a spike to find a way to solve.

**Pull request merged**